### PR TITLE
Refactor constants

### LINKED ISSUE #102: Refactor command words into constants
Every class that implements the `Command` should have a `COMMAND_WORD` and `FULL_COMMAND` which represents the String passed to the `@Command` annotation and the complete command that the user must enter to call this command.

For instance,
```java
@Command(name = RootCommand.COMMAND_WORD)
class RootCommand implements Command {
    public static final String COMMAND_WORD = "";
    public static final String FULL_COMMAND = COMMAND_WORD;
    // ...
}

@Command(name = AddCommand.COMMAND_WORD)
class AddCommand implements Command {
    public static final String COMMAND_WORD = "add";
    public static final String FULL_COMMAND = RootCommand.FULL_COMMAND + COMMAND_WORD;
    // ...
}
```

### PR DESCRIPTION
**Pull request by changjy44**
### Changes
1. Refactor all commands to use constant COMMAND_WORD and FULL_COMMAND.
2. Add equals method for all tests to include strong equality and stronger checks for nulls in optional arguments.
3. Add outline tests for all parsers (probably not important I will implement them in a separate PR).
4. Add aliases as static constants for all commands
5. Improve validation regex for links

Fixes #102

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/AssignCommand.java [hunk 1]
```diff
@@ -12,11 +12,15 @@
 /**
  * Command that contains all subcommands starting with {@code assign}.
  */
-@CommandLine.Command(name = "assign", aliases = {"as"}, mixinStandardHelpOptions = true, subcommands = {
-    AssignTaskCommand.class,
-    AssignTaskRandomlyCommand.class,
-})
+@CommandLine.Command(name = AssignCommand.COMMAND_WORD,
+        aliases = {"as"}, mixinStandardHelpOptions = true, subcommands = {
```
Comment (mjgui) [inline src/main/java/seedu/address/logic/commands/AssignCommand.java:16]: can we store the alias as a static constant also?

Comment (mjgui) [reply src/main/java/seedu/address/logic/commands/AssignCommand.java:16]: also this is unrelated to this issue but we should add aliases to our UG also

File: src/main/java/seedu/address/logic/commands/DeleteCommand.java [hunk 2]
```diff
@@ -35,4 +39,9 @@ public CommandResult execute(Model model) throws CommandException {
         throw new CommandException(String.format(MESSAGE_INVALID_COMMAND_WITH_HELP_FORMAT,
                 commandSpec.qualifiedName().trim()));
     }
+    @Override
```
Comment (ling-ui) [inline src/main/java/seedu/address/logic/commands/DeleteCommand.java:42]:
```suggestion

    @Override
```

**Review by ling-ui — APPROVED** [review]
LGTM 😃

**Review by mjgui — APPROVED** [review]

**Pull request merged**